_pages/doc/contributing-to-llvm.md

```diff
@@ -0,0 +1,102 @@
+---
+title: "Contributing to LLVM"
+---
+
+## Prerequisites
+
+First, read the [official documentation](https://llvm.org/docs/Contributing.html).
+
+## Setup
+
+### Forking and cloning the LLVM repository
+
+Head over to [the LLVM repository page](https://github.com/llvm/llvm-project) and fork it.
+
+Once GitHub has finished forking the repo, use git to clone your fork.
+
+```bash
+$ git clone git@github.com:<your-github-username>/llvm-project.git
+$ cd llvm-project
+$ mkdir build
+$ cd build
+```
+
+## Building
+
+See [How to Build M68k LLVM](build-from-source).
+
+## Running the M68k tests
+
+### Running tests using CMake
+
+#### Running all the tests
+
+Invoke CMake to run all the tests:
+
+```bssh
+cmake --build . --target check-all
+```
+
+#### Running specific sets of tests
+
+It takes significantly longer to run all the tests than it does to only run targeted M68k-related tests. In order to iterate more quickly, you can run only the M68k tests, rather than all the tests.
+
+```bash
+cmake --build . --target check-llvm-codegen-m68k
+```
+
+There are several useful test-related targets:
+
+| Target                            | Directory containing tests  |
+| --------------------------------- | --------------------------- |
+| `check-llvm-codegen-m68k`         | `test/CodeGen/M68k`         |
+| `check-llvm-debuginfo-m68k`       | `test/DebugInfo/M68k`       |
+| `check-llvm-mc-disassembler-m68k` | `test/MC/Disassembler/M68k` |
+| `check-llvm-mc-m68k`              | `test/MC/M68k`              |
+
+Additionally, the `llvm-test-depends` target builds all the dependencies required to run LLVM tests, but does not run the tests.
+
+### Running tests using `llvm-lit`
+
+Invoke CMake to build the testing dependencies:
+
+```bash
+cmake --build . --target llvm-test-depends
+```
+
+Invoke `llvm-lit` to run the M68k tests:
+
+```bash
+bin/llvm-lit \
+    ../llvm/test/CodeGen/M68k \
+    ../llvm/test/DebugInfo/M68k \
+    ../llvm/test/MC/Disassembler/M68k \
+    ../llvm/test/MC/M68k
```

Comment:
actually, you don't have to point to the test folders in the source tree, you can do this inside the _build folder_:
```
./bin/llvm-lit test/CodeGen/M68k ...
```
The reason we can do this is slightly complicated: there is actually a "local" LIT config in the build folder: `<build dir>/test/lit.site.cfg.py`, which loads the main LIT configs from the source tree along with the actual test files.